src/GcsBackup.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\ProjectBackup;
+
+use DateTimeImmutable;
+use Google\Cloud\Storage\StorageClient;
+use Google\Cloud\Storage\StorageObject;
+use Keboola\StorageApi\Client as StorageApi;
+use Psr\Log\LoggerInterface;
+
+class GcsBackup extends Backup
+{
+    private array $signedUrls = [];
+
+    public function __construct(
+        StorageApi $sapiClient,
+        readonly StorageClient $storageClient,
+        readonly string $bucketName,
+        readonly string $path,
+        readonly bool $generateSignedUrls = false,
+        ?LoggerInterface $logger = null,
+    ) {
+        parent::__construct($sapiClient, $logger);
+    }
+
+    /**
+     * @param resource|string $content
+     */
+    protected function putToStorage(string $name, $content): void
+    {
+        $bucket = $this->storageClient->bucket($this->bucketName);
+        $object = $bucket->upload($content, [
+            'name' => sprintf('%s%s', $this->path, $name),
+        ]);
+
+        if ($this->generateSignedUrls) {
+            $this->buildTreeFromPath($object, $name);
+        }
+    }
+
+    public function backupSignedUrls(): void
```

Comment:
k cemu to je? kazdopadne to nema testik :)

Reply:
pro zazálohování těch signedURLs.... je to specifikum pro GCS a bude se dělat jen při migraci - např pro data takeout se to dělat nemusí

test každopádně doplním